docs/docs-content/clusters/public-cloud/azure/required-permissions.md

```diff
@@ -0,0 +1,366 @@
+| Microsoft.Network/publicIPAddresses/delete           | Subscription |
+| Microsoft.Network/publicIPAddresses/join/action      | Subscription |
+| Microsoft.Network/publicIPAddresses/read             | Subscription |
+| Microsoft.Network/publicIPAddresses/write            | Subscription |
+| Microsoft.Network/routeTables/delete                 | Subscription |
+| Microsoft.Network/routeTables/read                   | Subscription |
+| Microsoft.Network/routeTables/write                  | Subscription |
+| Microsoft.Network/virtualNetworks/delete             | Subscription |
+| Microsoft.Network/virtualNetworks/join/action        | Subscription |
+| Microsoft.Network/virtualNetworks/join/action        | Subscription |
+| Microsoft.Network/virtualNetworks/joinLoadBalancer/action | Subscription |
+| Microsoft.Network/virtualNetworks/peer/action         | Subscription |
+| Microsoft.Network/virtualNetworks/read               | Subscription |
+| Microsoft.Network/virtualNetworks/subnets/delete     | Subscription |
+| Microsoft.Network/virtualNetworks/subnets/join/action| Subscription |
+| Microsoft.Network/virtualNetworks/subnets/joinLoadBalancer/action | Subscription |
+| Microsoft.Network/virtualNetworks/subnets/read       | Subscription |
+| Microsoft.Network/virtualNetworks/subnets/virtualMachines/read | Subscription |
+| Microsoft.Network/virtualNetworks/subnets/write      | Subscription |
+| Microsoft.Network/virtualNetworks/virtualMachines/read | Subscription |
+| Microsoft.Network/virtualNetworks/virtualNetworkPeerings/delete | Subscription |
+| Microsoft.Network/virtualNetworks/virtualNetworkPeerings/read | Subscription |
+| Microsoft.Network/virtualNetworks/virtualNetworkPeerings/write | Subscription |
+| Microsoft.Network/virtualNetworks/write              | Subscription |
+| Microsoft.Resources/subscriptions/resourceGroups/read | Subscription |
+| Microsoft.Storage/storageAccounts/blobServices/containers/read | Subscription |
+| Microsoft.Storage/storageAccounts/blobServices/containers/write | Subscription |
+| Microsoft.Storage/storageAccounts/blobServices/listKeys/action | Subscription |
+| Microsoft.Storage/storageAccounts/read                | Subscription |
+| Microsoft.Storage/storageAccounts/write               | Subscription |
+
+
+
+
+## Validator
+
+To use the [Validator](https://github.com/spectrocloud-labs/validator) with the [Azure plugin](https://github.com/spectrocloud-labs/validator-plugin-azure) you need the following actions to be able to ensure the roles attached to the service principal used by Validator have the required permissions:
+
+- Microsoft.Authorization/denyAssignments/read
+- Microsoft.Authorization/roleAssignments/read
+- Microsoft.Authorization/roleDefinitions/read
+
+
+You can use the Palette CLI to get started with the Validator. Check out the [Validator](../../../palette-cli/commands/validator.md) command reference for more information.
```

Comment:
When I see this section, I assume that users looking for actions here already know what the Validator is. So, this sentence feels a little out of place here.